docs/contributing/commit-discipline.md

```diff
@@ -86,87 +86,224 @@ check out GitHub's excellent [blog post](https://github.blog/2022-06-30-write-be
+- If it's hard to find a 1-2 word description of the part of the code base
+  affected by your commit, consider again whether you have structured your
+  commits well.
+- Never use a generic term like "bug", "fix", or "refactor".
+
+### Commit summary, part 2
+
+This is a **complete sentence** that briefly summarizes your changes. There are
+a few rules to keep in mind:
+
+- Start the sentence with an
+  [imperative](https://en.wikipedia.org/wiki/Imperative_mood) verb, e.g.
+  "fix", "add", "change", "rename", etc.
+- Use proper capitalization and punctuation.
+- Avoid abbreviations and acronyms.
+- Be concise, and don't include unnecessary details. For example, "Change X and
+  update tests/docs," would be better written as just, "Change X," since (as
+  discussed above) _every_ commit is expected to update tests and documentation
+  as needed.
+- Make it readable to someone who is familiar with Zulip's code base, but hasn't
+  been involved with the project you're working on.
+- Use no more than 76 characters for the entire commit summary (parts 1 and 2).
+
+### Examples of good commit summaries
+
+- `provision: Improve performance of installing npm.`
+- `channel: Discard all HTTP responses while reloading.`
+- `integrations: Add GitLab integration.`
+- `typeahead: Rename compare_by_popularity() for clarity.`
+- `typeahead: Convert to ES6 module.`
+- `tests: Compile Handlebars templates with source maps.`
+- `blueslip: Add feature to time common operations.`
+- `gather_subscriptions: Fix exception handling bad input.`
+- `stream_settings: Fix save/discard widget on narrow screens.`
+
+#### Detailed example
+
+- **Good summary**: "gather_subscriptions: Fix exception handling bad input."
+- **Not so good alternatives**:
+  - "gather_subscriptions was broken": This doesn't explain how it was broken, and
+    doesn't follow the format guidelines for commit summaries.
+  - "Fix exception when given bad input": It's impossible to tell what part of the
+    codebase was changed.
+  - Not using the imperative:
+    - "gather_subscriptions: Fixing exception when given bad input."
+    - "gather_subscriptions: Fixed exception when given bad input."
+
+### Commit description
+
+The body of the commit message should explain why and how the change
+was made. Like a good code comment, it should provide context and
+motivation that will help both a reviewer now, and a developer looking
+at your changes a year later, understand the motivation behind your
+decisions.
+
+Many decisions may be documented in multiple places (for example, both
+in a commit message and a code comment). The general rules of thumb are:
+
+- If the information will be helpful long-term for someone trying to
+  read and understand the code base, include it in a code comment.
+- If the information will be helpful long-term for someone trying to
+  investigate why these changes are a complete and correct
+  implementation of the commit's idea, include it in the commit
+  message.
```

Comment:
This part I feel could be a bit clearer on the distinction between these two concepts. Here's a version adapted from what I have [in the zulip-mobile style guide](https://github.com/zulip/zulip-mobile/blob/main/docs/style.md#commit-messages-code):
```suggestion
- Use the commit message for information that's relevant for someone
  trying to understand the change this commit is making, or the difference
  between the old version of the code and the new version. In particular,
  this includes information about why the new version of the code is better than,
  or not worse than, the old version.
- Use code comments, or the code itself, for information that's relevant
  for someone trying to read and understand the new version of the code
  in the future, without comparing it to the old version.
```

Reply:
That looks great to me.

Reply:
Yeah that's great!